### PR TITLE
Add support for server timing response propagator

### PR DESCRIPTION
**Pull request by pmcollins**

### CODE REVIEW

File: src/splunk_otel/propagator.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        Server-Timing: traceparent;desc="00-e899d68fca52b66d3facae0bdaf764db-159efb97d6b56568-01"
+        Access-Control-Expose-Headers: Server-Timing
+        """
+        span = trace.get_current_span(context)
+        span_context = span.get_span_context()
+        if span_context == trace.INVALID_SPAN_CONTEXT:
+            return
+
+        header_name = "Server-Timing"
+        flags = span_context.trace_flags
```
Comment (johnbley) [inline src/splunk_otel/propagator.py:48]: I believe our rum clients specifically look for `-01` for the flags.  Best to just hardcode in the `f'`.

**Review by johnbley — CHANGES_REQUESTED** [review]

File: tests/test_propagator.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    ctx = trace.set_span_in_context(span)
+    prop = ServerTimingResponsePropagator()
+    carrier = {}
+    prop.inject(carrier, ctx)
+    assert carrier["Access-Control-Expose-Headers"] == "Server-Timing"
+    assert carrier["Server-Timing"] == 'traceparent;desc="00-00000000000000000000000000000001-0000000000000002-01"'
+
+
+def test_inject_not_sampled():
```
Comment (johnbley) [inline tests/test_propagator.py:38]: This one could be dropped or kept with a change in output.

File: src/splunk_otel/propagator.py (hunk 1 above)
Comment (pmcollins) [reply src/splunk_otel/propagator.py]: Sure, I've hardcoded the `-01` -- happy to change if that's not what you're suggesting.

File: tests/test_propagator.py (hunk 2 above)
Comment (pmcollins) [reply tests/test_propagator.py]: Removed now that the 01 flag is hardcoded.

**Review by johnbley — APPROVED** [review]

**Pull request merged**